src/main/java/org/osc/controller/nsfc/api/NeutronSfcSdnRedirectionApi.java

```diff
@@ -246,25 +248,84 @@ public InspectionHookElement getInspectionHook(String inspectionHookId) throws E
 
     // SFC methods
     @Override
-    public NetworkElement registerNetworkElement(List<NetworkElement> inspectedPorts) throws Exception {
-        return null;
+    public NetworkElement registerNetworkElement(List<NetworkElement> inspectionPorts) throws Exception {
+       
+        ServiceFunctionChainEntity sfc = new ServiceFunctionChainEntity();
+        //check for null or empty list
+        this.utils.throwExceptionIfNullOrEmptyNetworElementList(inspectionPorts, "PortPairGroup memeber list");
+     
+        this.utils.addPortPairGroupToServiceFunction(inspectionPorts, sfc);
+        return this.txControl.required(() -> {
+            this.em.merge(sfc);
+            return this.utils.sendServiceFunctionChainId(sfc);
+        });
+       
     }
 
     @Override
-    public NetworkElement updateNetworkElement(NetworkElement portGroup, List<NetworkElement> inspectedPorts)
+    public NetworkElement updateNetworkElement(NetworkElement portGroup, List<NetworkElement> inspectionPorts)
             throws Exception {
-        // no-op
-        return null;
+        
+        this.utils.throwExceptionIfNullElementAndParentId(portGroup, "PortPairGroup ServiceFunctionChain Id");
```

Comment:
why do we check parent id? we should just check element id. NetworkElement portGroup -> is actually sfc

Reply:
Was expecting sfc_id in parent...modified to element-id now.